snovault/tests/test_schema_utils.py

```diff
@@ -364,3 +365,58 @@ def test_schema_utils_resolve_merge_ref_in_embedded_schema(testapp):
     embedded = testapp.get(f'/{atid}?frame=embedded', status=200).json
     assert embedded['link']['quality'] == 5
     assert embedded['link']['linked_targets'][0]['test_description'] == 'test'
+
+
+@pytest.mark.parametrize('invalid_date', [
+    'not a date',
+    'also-not-a-date',
+    '10-5-2022',
+    '10-05-almost',
+    '10-05-2023f'
+])
+def test_schema_utils_validates_dates(testapp, invalid_date):
+    """ Tests that our validator will validate dates """
+    schema = {
+        "type": "object",
+        "properties": {
+            "date_property": {
+                "type": "string",
+                "format": "date"
+            }
+        }
+    }
+    _, errors = validate(schema, {
+        'date_property': invalid_date
+    })
+    date_error = str(errors[0])
+    assert f"'{invalid_date}' is not a 'date'" in date_error
```

Comment:
The error message shows the offending string by calling `repr` or using `%r` or `{...!r}`, and it makes more sense to me to do the same thing here. The difference is subtle, but if you had an invalid date with a single quote mark in it, the way it would show up in the error message would involve escaping (presumably, or the use of doublequotes instead) and not just the addition of single quotes on the outside.
```suggestion
    assert f"{invalid_date!r} is not a 'date'" in date_error
```